### PR TITLE
Apply the detected changes

### LINKED ISSUE #6: [pip] Is there any way to apply the results to the project files?
I mean, it would be nice to provide a way to use the output (list of unneeded libraries) in order to automatically apply the changes (remove them from `build-depends` list) in the `.cabal` or `.stack` project files.

### PR DESCRIPTION
**Pull request by dfithian**
Fixes #6 

Apply the proposed changes.

If you specify `--apply`, it will ask for confirmation after printing the results for every target.
If you also specify `--no-verify`, it will apply the changes without confirmation. Haven't actually tested this part yet, but that's the way it _should_ work.

One caveat here is that the pretty printing of the package description is different than the input file in a lot of cases, so it _will_ mess up the .cabal file in question. Also I made it ignore package.yaml files. Maybe both of those things should be warnings.

Checklist:
- [x] Test `--apply`
- [x] Test `--no-verify`
- [x] Add warnings for changing `.cabal` file and ignoring `package.yaml`
- [x] If possible, instead of pretty printing the output, maybe do something like parsing the file halfway and modifying it in-place to cause as little churn as possible. Could do something like parsing `test-suite <name> *> indentation *> build-depends <* end-indentation`, overwriting the build-depends, and overwriting.
- [x] Fix up the README

### CODE REVIEW

File: app/main.hs [hunk 1]
```diff
@@ -104,21 +120,19 @@ main = do
...
       let addSelf = if optsNoIgnoreSelf then id else Set.insert (T.DependencyName packageName)
       baseUsedDependencies <- fmap mconcat . for packageCompilables $ \compilable@T.Compilable {..} -> do
         usedDependencies <- addSelf <$> getCompilableUsedDependencies dependencyByModule compilable
         let (baseUsedDependencies, otherUsedDependencies) = Set.partition (flip Set.member packageBaseDependencies) usedDependencies
             otherUnusedDependencies = Set.difference compilableDependencies otherUsedDependencies
         unless (Set.null otherUnusedDependencies) $ do
-          liftIO . putStrLn . unpack $ "Some unused dependencies for " <> pack (show compilableType) <> " " <> T.unCompilableName compilableName <> " in package " <> packageName
-          traverse_ (liftIO . putStrLn . unpack . ("  " <>) . T.unDependencyName) $ Set.toList otherUnusedDependencies
-          put $ ExitFailure 1
+          shouldFail <- liftIO $ Unused.apply package otherUnusedDependencies (Just compilable) apply
```
Comment (dfithian) [inline app/main.hs:130]: This runs for each target so it's gonna overwrite... I think we need `State` or something here.

**Comment by dfithian** [comment]
As of the last commit, I tested it like this:

1. add `transformers` as a dependency to `prune-juice.cabal`
2. run `stack run -- prune-juice --apply apply-no-verify --apply-strategy smart`
3. it correctly removed the dependency from the file, however it slightly changed the format
Screen Shot 2021-11-08 at 2 01 29 PM

**Comment by dfithian** [comment]
@mirko-plowtech do you think this fixes your issue? You can test it out by running the following:

```bash
stack run -- prune-juice --apply
```

If you don't want it to prompt before applying, run

```bash
stack run -- prune-juice --no-verify
```

If it messes up your .cabal file, you can try running in "safe" mode with

```bash
stack run -- prune-juice --apply-strategy safe
```

**Comment by dfithian** [comment]
I attempted this on my company's cabal files, which use `common` stanzas, and it didn't apply any of the changes. So I'll try to parse the `common` stanzas.

**Comment by mirko-plowtech** [comment]
> @mirko-plowtech do you think this fixes your issue? You can test it out by running the following:
> 
> ```shell
> stack run -- prune-juice --apply
> ```
> 
> If you don't want it to prompt before applying, run
> 
> ```shell
> stack run -- prune-juice --no-verify
> ```
> 
> If it messes up your .cabal file, you can try running in "safe" mode with
> 
> ```shell
> stack run -- prune-juice --apply-strategy safe
> ```

Sorry the delay, I will try to test it this weekend =)

**Comment by dfithian** [comment]
I'm going to merge this so that some other contributions can be merged on top of it.

**Pull request merged**